### PR TITLE
Bug 1020757 - Handle generic error case for simpin r=Arthur

### PR DESCRIPTION
**Pull request by EragonJ**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=c4f18ed455f1)

File: apps/settings/js/panels/simpin_dialog/simpin_dialog.js [hunk 1]
```diff
@@ -4,10 +4,11 @@
  *
  * @module @SimPinDialog
  */
-define(function() {
+define(function(require) {
```
Comment (crh0716) [inline apps/settings/js/panels/simpin_dialog/simpin_dialog.js]: xd

Comment (EragonJ) [inline apps/settings/js/panels/simpin_dialog/simpin_dialog.js]: :hankey:

File: apps/settings/js/panels/simpin_dialog/simpin_dialog.js [hunk 2]
```diff
@@ -344,48 +345,82 @@ define(function() {
...
 
-      var msgId = (retryCount > 1) ? 'AttemptMsg3' : 'LastChanceMsg';
-      this._showMessage(lockType + 'ErrorMsg', lockType + msgId, {
-        n: retryCount
-      });
-      this._showRetryCount(retryCount);
+        if (lockType === 'pin' || lockType === 'fdn') {
+          this._elements.pinInput.value = '';
+          this._elements.pinInput.focus();
+        } else if (lockType === 'puk') {
```
Comment (crh0716) [inline apps/settings/js/panels/simpin_dialog/simpin_dialog.js]: We should also handle `lockType === 'puk2'`.

Comment (crh0716) [inline apps/settings/js/panels/simpin_dialog/simpin_dialog.js]: And in this case I believe both `newPinInput.value` and `confirmPinInput.value` should also be cleared.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bb219c9ae003ced776e9ef3a5628d596d887f27d

**Pull request merged**